hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/curator/ZKCuratorManager.java

```diff
@@ -503,4 +644,50 @@ private void setJaasConfiguration(ZKClientConfig zkClientConfig) throws IOExcept
       zkClientConfig.setProperty(ZKClientConfig.LOGIN_CONTEXT_NAME_KEY, JAAS_CLIENT_ENTRY);
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Helper class to contain the Truststore/Keystore paths for the ZK client connection over
+   * SSL/TLS.
+   */
+  public static class TruststoreKeystore{
+    private static String keystoreLocation;
+    private static String keystorePassword;
+    private static String truststoreLocation;
+    private static String truststorePassword;
+    /** Configuration for the ZooKeeper connection when SSL/TLS is enabled.
+     * When a value is not configured, ensure that empty string is set instead of null.
+     * @param conf ZooKeeper Client configuration
+     */
+    public TruststoreKeystore(Configuration conf){
+
+      keystoreLocation =
+          StringUtils.defaultString(conf.get(CommonConfigurationKeys.ZK_SSL_KEYSTORE_LOCATION,
```

Comment:
Why the StringUtils.defaultString is needed? 
I mean, conf.get() will return an empty string if the config is not found, given that you passed empty strings for all conf.get calls already.

Reply:
I was getting some NPE exception based on my recollection without this, but can't reproduce it now, so removing it and will see during the build if it comes up again.